python/nav/ipdevpoll/plugins/sensors.py

```diff
@@ -77,7 +77,6 @@ def handle(self):
                     "Found %d sensors from %s", len(all_sensors), type(mib).__name__
                 )
                 self._store_sensors(all_sensors)
-                break
```

Comment:
This was done so sensors are collected from all the MIBs listed in JUNIPER_NETWORKS_INC line in the config instead of stopping at the first one where it finds any sensors. This will impact all sensor MIBs, so I'm wondering 1) is there a reason why it should stop after finding sensors in one MIB? 2) If so, ideas for config that will allow only Juniper devices to collect sensors from multiple MIBs?

Reply:
Very good point. The juniper chassis error counts will also need to look into two different MIBs if we are to support all juniper equipment that has these counters.

Reply:
> 1) is there a reason why it should stop after finding sensors in one MIB?

I think the main reason is that some of the first devices this code was tested
on would report the same sensors in multiple MIBs (i.e. the sensors reported in
a proprietary MIB would also appear in `ENTITY-SENSOR-MIB`, and there is no
generic way of determining that these records represent the same thing).

> 2) If so, ideas for config that will allow only Juniper devices to collect
> sensors from multiple MIBs?

Naw. I think the original idea is a bit defeatist. If a device wants to report
multiple versions of the same sensors, that's not our problem, that's the
vendor's problem (and a little bit, the user's: Now ipdevpoll will spend twice
the time collecting data from this device, and twice the storage space to
persist it - but hey, storage is cheap these days).


So, IMO: It's a good idea to remove the `break` statement!

Reply:
@hmpf wrote:

> Very good point. The juniper chassis error counts will also need to look into
> two different MIBs if we are to support all juniper equipment that has these
> counters.

The point is good, but I don't see how it related to the ongoing alert count
collection feature. It's unlikely this will touch the sensors framework at all
(unless you plan to also keep time series data of alert counts).